### PR TITLE
Update the documentation

### PR DESCRIPTION
**Pull request by sanjaychoudhury**
Updated documentation in the following areas:
1. Updated README.md to include a catalog of supported Types
2. Updated README.md to include detailed usage documentation for each Type
3. Updated each Type's @doc instance variable with a brief description of the type. Updated the documentation properties associated with the Type using the 'desc' call. Content for the documentation are derived from the corresponding ACI Managed Object.

### CODE REVIEW

**Comment by rahushen** [comment]
👍

File: lib/puppet/type/cisco_aci_vzsubj.rb [hunk 1]
```diff
@@ -66,30 +75,32 @@ def self.title_patterns
...
   end
 
   newproperty(:descr) do
-    desc 'https://pubhub.devnetcloud.com/media/apic-mim-ref-311/docs/MO-vzSubj.html#descr'
+    desc 'Specifies the description of a policy component.'
   end
 
   newproperty(:name_alias) do
-    desc 'https://pubhub.devnetcloud.com/media/apic-mim-ref-311/docs/MO-vzSubj.html#nameAlias'
+    desc ''
```
Comment (mikewiebe) [inline lib/puppet/type/cisco_aci_vzsubj.rb]: IMO, we should fill in something for each property, even if it's simply `Specifies an alias for the managed object`.  Empty strings just look incomplete.

File: lib/puppet/type/cisco_aci_vzentry.rb [hunk 2]
```diff
@@ -66,62 +68,62 @@ def self.title_patterns
...
   end
 
   newproperty(:icmpv4_t) do
-    desc 'https://pubhub.devnetcloud.com/media/apic-mim-ref-311/docs/MO-vzEntry.html#icmpv4T'
+    desc ''
   end
 
   newproperty(:icmpv6_t) do
-    desc 'https://pubhub.devnetcloud.com/media/apic-mim-ref-311/docs/MO-vzEntry.html#icmpv6T'
+    desc ''
```
Comment (mikewiebe) [inline lib/puppet/type/cisco_aci_vzentry.rb]: Again here.  We could simply fill in `Specify ICMP v6 type` or something simple along those lines.

**Comment by rahushen** [comment]
@mikewiebe ... took care of the empty fields.

**Review by mikewiebe — APPROVED** [review]

**Review by rahushen — APPROVED** [review]

**Pull request merged**